src/captiveportal/captiveportalrequest.cpp

```diff
@@ -76,20 +76,28 @@ void CaptivePortalRequest::createRequest(const QUrl& url) {
           });
 
   connect(request, &NetworkRequest::requestCompleted,
-          [this](const QByteArray& data) {
+          [this, request](const QByteArray& data) {
             logger.log() << "Captive portal request completed:" << data;
 
             --m_pendingRequests;
             m_completed = true;
             deleteLater();
 
+            // Usually, captive-portal pages do a redirect to an internal page.
+            if (request->statusCode() != 200) {
+              logger.log() << "Captive portal detected. Expected 200, received:"
+                           << request->statusCode();
+              emit completed(true);
```

Comment:
Maybe we should also check that there really is something to redirect to e.g
```
!request->rawHeader("Location").isEmpty();
```

Reply:
What do you want to do with the location?